ledger/participant-integration-api/src/main/scala/platform/indexer/ha/PollingChecker.scala

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer.ha
+
+import java.util.{Timer, TimerTask}
+import java.util.concurrent.atomic.AtomicBoolean
+
+import akka.stream.KillSwitch
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+
+import scala.util.{Failure, Success, Try}
+
+class PollingChecker(
+    periodMillis: Long,
+    checkBlock: => Unit,
```

Comment:
no such block notion... I don't know where I got this, as a matter of fact I don't like it neither :smile: let's remove it

wrt Bool/Unit in the result: I would rather keep to the Unit, so no extra exception handling needed. Would like to follow the "whatever goes wrong is wrong, otherwise good" approach here. (this guy would have little room to maneuver if it gets an exception anyway)